lib/push-jenkins-update.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const githubClient = require('./github-client')
+
+const jenkinsToGhStatusMap = {
+  'SUCCESS': {
+    state: 'success',
+    message: 'all tests passed'
+  },
+  'FAILURE': {
+    state: 'failure',
+    message: 'build failure'
+  },
+  'ABORTED': {
+    state: 'error',
+    message: 'build aborted'
+  },
+  'UNSTABLE': {
+    state: 'error',
+    message: 'build unstable'
+  }
+}
+
+const rxDisplayName = /node\-test\-commit\-(\w+)/
+
+function pushJenkinsUpdate (options, build) {
+  const statusOpts = extendWithBuildData(options, build)
+  const createGhStatus = createGhStatusFn(statusOpts)
+  const isPending = build.result === null
+  const matchedGhStatus = jenkinsToGhStatusMap[build.result]
+
+  if (isPending) {
+    createGhStatus('pending', 'build in progress')
```

Comment:
I'd like this to be a bit more flexible. We could even post updates between compiling and testing. Linters aren't necessarily "building" either. Perhaps having a set of defaults or pass on what is posted through curl?